src/time/Waveform.cpp

```diff
@@ -88,11 +98,11 @@ Eigen::VectorXd Waveform::extrapolateData(int order, int timeWindows)
 Eigen::VectorXd Waveform::interpolateData(int order, int timeWindows, double dt)
 {
   Eigen::VectorXd interpolatedValue;
-  if (order == 0) {
+  if ((order == 0) || (timeWindows < 2 && order > 0)) {
```

Comment:
This is an important change in the interpolation: If `timeWindows == 1`, only have a single sample available and therefore only use constant interpolation based on that sample. This means that currently the first time window will always use constant interpolation. This can be fixed in the future by providing initial values at the beginning and end of the first window.